quokka_editor_front/src/components/organisms/documentsEditor/handlers.ts

```diff
@@ -30,9 +32,17 @@ export const onBeforeChangeHandler = (
   value: string,
   socket: React.MutableRefObject<WebSocket | null>,
   client: ClientState,
-  setClient: React.Dispatch<React.SetStateAction<ClientState>>
+  setClient: React.Dispatch<React.SetStateAction<ClientState>>,
+  document: DocumentState
 ) => {
   if (!socket.current) return;
+  if (!canEdit(document)) {
+    toast.error(
+      "You don't have permission to edit this document",
+      TOAST_OPTIONS
+    );
+    return;
+  }
```

Comment:
Shouldn't we add navigation to document lists or if it is anonymous to the login page?

Reply:
No, because it only checks user's ability to edit the document. They can read the document when the document is shared, but they aren't able to edit it if privileges don't allow it